web/src/left_sidebar_navigation_area.js

```diff
@@ -1,12 +1,53 @@
 import $ from "jquery";
 
+import render_active_view_row from "../templates/active_view_row.hbs";
+
 import * as pm_list from "./pm_list";
 import * as resize from "./resize";
 import * as scheduled_messages from "./scheduled_messages";
+import * as starred_messages from "./starred_messages";
 import * as ui_util from "./ui_util";
+import * as unread from "./unread";
 
 let last_mention_count = 0;
 let left_sidebar_navigation_area_expanded = true;
+const views_data = {
```

Comment:
Yeah, I've been playing with and thinking about this, but it's not the way to go. This introduces a whole lot of tricky code duplication with the ordinary rows in `left_sidebar.hbs`.

I think a better way to do this would be to create partials for each of these rows. The could then be loaded up at compile time within `left_sidebar.hbs`, but also then accessed independently for the active row.

That said, there are some other structural things I'm pondering here, so I'll follow back up if my thinking on this changes.